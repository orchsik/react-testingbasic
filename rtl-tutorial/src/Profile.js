const Profile = ({ username, name }) => {
  return (
    <div>
      <b>{username}</b>
      <span>{name}</span>
    </div>
  );
};

export default Profile;
