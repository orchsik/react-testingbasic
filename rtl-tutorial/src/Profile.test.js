import { render } from "@testing-library/react";
import Profile from "./Profile";

describe("<Profile>", () => {
  const props = {
    username: "액션가면",
    name: "신짱구",
  };
  const utils = render(<Profile {...props} />);

  it("matches sanpshot", () => {
    expect(utils.container).toMatchSnapshot();
  });
  it("shows teh props correctly", () => {
    utils.getByText("액션가면");
    utils.getByText("신짱구");
    utils.getByText(/신/);
  });
});
