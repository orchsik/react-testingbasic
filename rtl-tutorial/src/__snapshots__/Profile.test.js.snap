// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile> matches sanpshot 1`] = `
<div>
  <div>
    <b>
      액션가면
    </b>
    <span>
      신짱구
    </span>
  </div>
</div>
`;
